### PR TITLE
Seeg fix

### PR DESCRIPTION
**Pull request by jaeilepp**
Takeover from https://github.com/mne-tools/mne-python/pull/2550

### CODE REVIEW

File: mne/tests/test_epochs.py [hunk 1]
```diff
@@ -1790,4 +1792,13 @@ def test_add_channels():
...
 
 
+def test_seeg():
+    """Test the compatibility of the Epoch object with SEEG data."""
+    n_epochs, n_channels, n_times, sfreq = 5, 10, 20, 1000.
+    data = np.random.randn(n_epochs, n_channels, n_times)
+    events = np.array([np.arange(n_epochs), [0] * n_epochs, [1] * n_epochs]).T
+    info = create_info(n_channels, sfreq, 'seeg')
+    epochs = EpochsArray(data, info, events)
+    assert epochs.average()
```
Comment (agramfort) [inline mne/tests/test_epochs.py]: you're not testing much here.

also never use assert but assert_true from nose.

File: mne/epochs.py [hunk 2]
```diff
@@ -735,7 +735,7 @@ def _compute_mean_or_stderr(self, picks, mode='ave'):
         # pick channels
         if picks is None:
             picks = pick_types(evoked.info, meg=True, eeg=True, ref_meg=True,
-                               stim=False, eog=False, ecg=False,
+                               stim=False, eog=False, ecg=False, seeg=True,
                                emg=False, exclude=[])
```
Comment (agramfort) [inline mne/epochs.py]: would there be a way to factorize this?

maybe a

DATA_PICKS = dict(eeg=True, meg=True ...)

in picks.py

and then in the files

picks = pick_types(info, **DATA_PICKS)

Comment (jaeilepp) [inline mne/epochs.py]: How about a separate convenience function like ``pick_data_channels``?

Comment (agramfort) [inline mne/epochs.py]: if you want but make it private

File: mne/epochs.py [hunk 3]
```diff
@@ -536,13 +536,13 @@ def _detrend_offset_decim(self, epoch, verbose=None):
...
-                               ref_meg=False, eog=False, ecg=False,
+                               ref_meg=False, eog=False, ecg=False, seeg=True,
                                emg=False, exclude=[])
             epoch[picks] = detrend(epoch[picks], self.detrend, axis=1)
 
         # Baseline correct
         picks = pick_types(self.info, meg=True, eeg=True, stim=False,
-                           ref_meg=True, eog=True, ecg=True,
+                           ref_meg=True, eog=True, ecg=True, seeg=True,
                            emg=True, exclude=[])
```
Comment (jaeilepp) [inline mne/epochs.py]: BTW. What are these reference MEG channels used for? Here they are not detrended, but baselining is applied.

Comment (agramfort) [inline mne/epochs.py]: they are used in CTF systems. They are basically magnetometers a bit further away from the head.

Comment (agramfort) [inline mne/epochs.py]: not detrending them is certainly a bug

File: mne/epochs.py (hunk 2 above)
Comment (jaeilepp) [inline mne/epochs.py]: Actually this is almost the default behavior of ``pick_types``. The convenience function would basically be ``pick_types(info, eeg=True, seeg=True)``. I was thinking of factoring by just changing the calls to that, but I feel it is less readable.

Comment (jaeilepp) [inline mne/epochs.py]: On a second thought, maybe there should be a convenience function that set the channels explicitly. The question is that should it use ``'auto'`` or ``True`` for ``ref_meg``?

Comment (Eric89GXL) [inline mne/epochs.py]: +1 on private `_pick_data_channels(exclude='bads')` that just wraps to `pick_types(info, meg=True, eeg=True, seeg=True, exclude=exclude)`. That way if we add a new channel type of some sort, we only change it in one place (DRY).

Comment (Eric89GXL) [inline mne/epochs.py]: I don't know about the ref stuff. I actually don't know the difference between `'auto'` and `True`... @agramfort any idea?

Comment (agramfort) [inline mne/epochs.py]: hum where is "auto" used.

Basically None is the default "smart" option. So for averaging you need the ref_meg as you'll need it for inverse modeling. But you don't need ref_meg for plotting or TFR or PSD ...

makes sense?

Comment (jaeilepp) [inline mne/epochs.py]: ``'auto'`` is the default for ``pick_types`` and the docs say:
```
If True include CTF / 4D reference channels. If 'auto', the reference
        channels are only included if compensations are present.
```
So I'm asking if the ``_pick_data_channels`` should include them or use ``auto``.

Comment (agramfort) [inline mne/epochs.py]: do what is the more DRY while not breaking API.

**Comment by agramfort** [comment]
appveyor error is related to GUI

I am restarting travis

could this pick data channel function be used else where?

maybe in TFR code?

**Comment by jaeilepp** [comment]
Does anyone have an idea why the filter test keeps failing here? I don't think any of the changes are related and I can't reproduce the failure on any of my environments.

**Comment by Eric89GXL** [comment]
@jaeilepp are you rebased to latest `master` here? It looks like it, just want to double-check.

Do you have a working Anaconda virtual environment? That might help you see the same set of errors. I have one I will check...

File: mne/tests/test_epochs.py [hunk 4]
```diff
@@ -1435,6 +1435,8 @@ def test_drop_epochs_mult():
 
 def test_contains():
     """Test membership API"""
+    # TODO This raw data does not contain seeg. The test cannot completely test
+    # membership.
```
Comment (Eric89GXL) [inline mne/tests/test_epochs.py]: It should only take a few lines to add some fake channels, see here for an example of adding ECG to a raw instance that doesn't have it already:

https://github.com/mne-tools/mne-python/blob/master/mne/simulation/tests/test_raw.py#L68

**Comment by jaeilepp** [comment]
I have environments for python 3.4 and 2.7.

**Comment by jaeilepp** [comment]
And i've rebased at least a couple of times, but the error persists. I'm not sure this is on the latest.

**Comment by Eric89GXL** [comment]
I also can't reproduce locally. I hate to say this, but at this point I'd debug by undoing sets of changes until I found the culprit. :( It should take about 10 minutes per push for the `nodata` build, which fails, to run. And you can cancel all the other builds as they come up on Travis.

Or take the eaiser way out, and assume it's related to this:

https://github.com/ContinuumIO/anaconda-issues/issues/479

And thus add a skip like this, only change the conditionals as necessary to fit maybe numpy 1.10 or above...?:

https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/tests/test_maxwell.py#L123

**Comment by agramfort** [comment]
how can a change of pick function make a test fail???

**Comment by Eric89GXL** [comment]
I don't know. Maybe some channel that was being filtered before isn't anymore or something. But I looked at the code and I don't see anywhere that should do that. That's why the first change I would test is reverting everything in one additional commit, to see if that even passes. Maybe Anaconda changed something that broke our tests, and it's just manifesting in this PR.

**Comment by agramfort** [comment]
I am afraid it's the way to go :(

**Comment by jaeilepp** [comment]
Seems like the filter test passes now even though it's in an identical state to the earlier commit, which failed. I hope the rendering failure is not here to stay.

**Comment by jaeilepp** [comment]
It's weird. Looks like the test caused the filter test to fail. But seems like it fails on a different test on every run...

**Comment by jaeilepp** [comment]
Ok. The filter test no longer fails and I think I know what was the issue. Earlier the testing data was created with numpys random module. This must have changed the random number generators state, which affected the values in filter test. Lesson to take home: it is dangerous to use random module when creating data for the tests.

**Comment by Eric89GXL** [comment]
We usually use np.random.RandomState to avoid such bugs, sorry you got hit
by one

**Comment by agramfort** [comment]
can you see if the _pick_data_channels could also be used elsewhere?

besides the travis error LGTM

**Comment by Eric89GXL** [comment]
I don't see any fix for the `filter` tests that use `np.random.rand*` instead of the `rng = np.random.RandomState(); rng.rand*` pattern. Can you either fix these as part of this PR, or open an issue so we don't forget to do it to avoid future Heisenbugs?

**Comment by agramfort** [comment]
see psd.py : compute_epochs_psd
ar.py : fit_iir_model_raw

**Comment by agramfort** [comment]
I just did

`git grep picks=None`

**Comment by jaeilepp** [comment]
> see psd.py : compute_epochs_psd
ar.py : fit_iir_model_raw

In these ref_meg channels are not included.

**Comment by agramfort** [comment]
ok. ping me when I shall review.

**Comment by jaeilepp** [comment]
I updated the random states to tests. Let's see if they pass.

**Comment by agramfort** [comment]
travis is not happy :(

**Comment by jaeilepp** [comment]
It fails on file download. I suppose a restart will fix it.

**Comment by agramfort** [comment]
let me know when travis is green

**Comment by Eric89GXL** [comment]
@jaeilepp you should be able to restart the builds yourself now that you have admin access, but I've done it this time

**Comment by jaeilepp** [comment]
How do you do that?

**Comment by Eric89GXL** [comment]
Do you see a little restart circle in the upper right of this page?

https://travis-ci.org/mne-tools/mne-python/jobs/91540326

**Comment by Eric89GXL** [comment]
@agramfort tests pass

**Comment by jaeilepp** [comment]
> Do you see a little restart circle in the upper right of this page?

I needed to sign up for travis. Now it's there. Thanks.

**Comment by Eric89GXL** [comment]
Changes LGTM, +1 for merge. Thanks for fixing the static RandomState stuff.

**Comment by agramfort** [comment]
thanks @jaeilepp !

**Pull request merged**